### PR TITLE
Update EigenDA contracts

### PR DESCRIPTION
**Pull request by teddyknox**

### CODE REVIEW

File: docs/eigenda/deployed-contracts.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+## EigenDA Contracts
+
+For more information on the contracts and their technical specifications, please see:
+[EigenLayer Middleware Docs](https://github.com/Layr-Labs/eigenlayer-middleware/tree/m2-mainnet/docs).
+
+### Current Mainnet Deployment
+
+No contracts have been deployed to mainnet yet.
+
+### Current Testnet Deployment
```
Comment (wadealexc) [inline docs/eigenda/deployed-contracts.md:15]: Please add branch/release context like the text removed in the other doc :heart:

**Review by wadealexc — CHANGES_REQUESTED** [review]
See feedback above!

File: docs/eigenda/deployed-contracts.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+|--------------------------------|------------------------------------------------------------------------------------------------------------|---------------------------------------------------------------|---------------------------------------------------------------|------------------------------------------------------------------------------------------------------------------|
+| RegistryCoordinator            | [`BLSRegistryCoordinatorWithIndices.sol`](https://github.com/Layr-Labs/eigenlayer-middleware/blob/v0.1.0-m2-goerli/src/BLSRegistryCoordinatorWithIndices.sol) | [`0x7661dA87296c0A245d11ADeD7f48265463bc3eC6`](https://goerli.etherscan.io/address/0x7661dA87296c0A245d11ADeD7f48265463bc3eC6) | [`0x7dfE0C25483E420eccD04eab8Be44c75a0CE82B5`](https://goerli.etherscan.io/address/0x7dfE0C25483E420eccD04eab8Be44c75a0CE82B5) | Proxy: [OpenZeppelin TUP@4.7.1](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/v4.7.1/contracts/proxy/transparent/TransparentUpgradeableProxy.sol) |
```
Comment (ian-shim) [inline docs/eigenda/deployed-contracts.md:19]: Looks like the links to solidity files are from `v0.1.0-m2-goerli` branch.
The branch we deployed yesterday was from `m2-mainnet-fixes`

File: docs/eigenda/deployed-contracts.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+| BLSApkRegistry                 | [`BLSPubkeyRegistry.sol`](https://github.com/Layr-Labs/eigenlayer-middleware/blob/v0.1.0-m2-goerli/src/BLSPubkeyRegistry.sol)                   | [`0xb406d8e5F4BA1430D57c8231cc872C2c369E1F5F`](https://goerli.etherscan.io/address/0xb406d8e5F4BA1430D57c8231cc872C2c369E1F5F) | [`0x7f57f298b7270B1D4C8964E33eB6eEfF31aD2063`](https://goerli.etherscan.io/address/0x7f57f298b7270B1D4C8964E33eB6eEfF31aD2063) | Proxy: [OpenZeppelin TUP@4.7.1](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/v4.7.1/contracts/proxy/transparent/TransparentUpgradeableProxy.sol) |
```
Comment (ian-shim) [inline docs/eigenda/deployed-contracts.md:22]: This should be the right solidity file for BLSApkRegistry: https://github.com/Layr-Labs/eigenlayer-middleware/blob/m2-mainnet-fixes/src/BLSApkRegistry.sol

**Review by ian-shim — DISMISSED** [review]

File: docs/eigenda/deployed-contracts.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+For more information on the contracts and their technical specifications, please see:
+[EigenLayer Middleware Docs](https://github.com/Layr-Labs/eigenlayer-middleware/tree/m2-mainnet/docs).
+
+### Current Mainnet Deployment
+
+No contracts have been deployed to mainnet yet.
+
+### Current Testnet Deployment
+
+The current testnet deployment of EigenDA is [`v0.3.0`](https://github.com/Layr-Labs/eigenda/releases/tag/v0.3.0) part of which includes middleware contracts at version [`v0.1.1-goerli-m2`](https://github.com/Layr-Labs/eigenlayer-middleware/tree/testnet) release. Below are the deployed contract addresses and links to source.
```
Comment (wadealexc) [inline docs/eigenda/deployed-contracts.md:17]: "version [v0.1.1-goerli-m2](https://github.com/Layr-Labs/eigenlayer-middleware/tree/testnet) release" wording is a little bit awkward, but not blocking

**Review by wadealexc — APPROVED** [review]
LGTM

**Pull request merged**